draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -391,6 +391,35 @@ DNSName values use the "preferred name syntax" as specified by {{Section 3.5 of
 
 Names in a `dns` claim represent the exact DNS name specified. Names in a `dns_wildcard` claim represent wildcard DNS names and are processed as if prepended with the string "`*.`" and then following the steps in {{Section 6.3 of !I-D.draft-ietf-uta-rfc6125bis}}.
 
+### Canonical order {#dnsname-order}
+
+The values in `dns_names` MUST be sorted lexicographically by label
+from right-to-left; that is:
+
+We consider `domain1 <= domain2`, if and only if
+
+1. `label1 < label2` lexicographically, or
+
+2. `label1 = label2`, and `subdomain1 <= subdomain2`,
+
+where `label1` is a label and `subdomain1` is such that either
+
+1. `domain1 = subdomain1.label1`, or
+
+2. `domain1 = label1`, and in that case `subdomain1` is the emty string;
+
+and `label2` and `subdomain2` are defined similarly  for `domain2`.
+
+The following is a list of properly sorted names in this order:
+
+~~~
+example.com
+b.example.com
+b.b.example.com
+ba.example.com
+z.example.com
+~~~
```

Comment:
This list is chosen such that normal sorting or sorting the reversed words will not work. Also it catches all cases in the definition.